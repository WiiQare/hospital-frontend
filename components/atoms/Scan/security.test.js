import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import Security from "./security";
import { StepContext } from ".";
import { SessionProvider } from "next-auth/react";
import { QueryClientProvider, QueryClient } from "react-query";

describe("Security", () => {
  let component;
  beforeEach(() => {
    const queryClient = new QueryClient();
    const res = render(
      <SessionProvider
        session={{
          user: {
            data: {
              access_token: "token123",
              providerId: "id123",
            },
          },
        }}
      >
        <QueryClientProvider client={queryClient}>
          <StepContext.Provider value={{ step: 1, setStep: () => {} }}>
            <Security shorten={true} />
          </StepContext.Provider>
        </QueryClientProvider>
      </SessionProvider>
    );
    component = res.container;
  });

  it("should render the component", () => {
    expect(component).toMatchSnapshot();
  });
});
