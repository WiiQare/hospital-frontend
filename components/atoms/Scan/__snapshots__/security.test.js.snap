// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Security should render the component 1`] = `
<div>
  <div
    class="h-full flex justify-center items-center pb-40"
  >
    <div
      class="w-full  mx-auto max-w-md p-6 overflow-hidden text-left align-middle transition-all transform bg-white shadow-sm rounded-xl space-y-8 mt-2"
    >
      <h1
        class="text-center font-semibold text-xl text-gray-700"
      >
        Entrez le code de sécurité
      </h1>
      <div
        class="space-y-3"
      >
        <div
          class="flex flex-col gap-2 w-full"
        >
          <form
            class="flex flex-col w-full justify-between items-center gap-3"
            id="form-security"
          >
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root w-full placeholder:text-gray-400 hover:outline-none focus:ring-0 border border-gray-300 rounded-lg focus:ring-sky css-wb57ya-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for=":r0:"
                id=":r0:-label"
              >
                Entrez le code de sécurité
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  id=":r0:"
                  name="securityCode"
                  placeholder="Entrez le code de sécurité"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-yjsfm1"
                  >
                    <span>
                      Entrez le code de sécurité
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </form>
        </div>
        <div
          class="flex flex-row-reverse gap-4"
        >
          <button
            class="bg-orange text-md py-2 px-4 rounded-lg effect-up text-white"
            form="form-security"
            type="submit"
          >
            Envoyer
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
